src/docker/blockchain/bitcoin.rs

```diff
@@ -58,36 +59,95 @@ impl Image for BitcoinNode {
             panic!("Internal Error: Url for bitcoin client should have been set.");
         });
 
-        let rpc_client = bitcoincore_rpc::Client::new(
-            endpoint.clone(),
-            bitcoincore_rpc::Auth::UserPass(Self::USERNAME.to_string(), Self::PASSWORD.to_string()),
-        )
-        .expect("Could not create client");
+        Self {
+            username: Self::USERNAME.to_string(),
+            password: Self::PASSWORD.to_string(),
+            endpoint,
+        }
+    }
+}
+
+#[derive(Debug, serde::Serialize)]
+pub struct GenerateQuery {
+    jsonrpc: String,
+    id: String,
+    method: String,
+    params: Vec<u32>,
+}
+
+impl GenerateQuery {
+    pub fn new(number: u32) -> Self {
+        GenerateQuery {
+            jsonrpc: "1.0".to_string(),
+            id: "generate".to_string(),
+            method: "generate".to_string(),
+            params: vec![number],
+        }
+    }
+}
 
-        Self { rpc_client }
+#[derive(Debug, serde::Serialize)]
+struct FundQuery {
+    jsonrpc: String,
+    id: String,
+    method: String,
+    params: Vec<String>,
+}
+
+impl FundQuery {
+    fn new(address: Address, amount: Amount) -> Self {
+        FundQuery {
+            jsonrpc: "1.0".to_string(),
+            id: "fund".to_string(),
+            method: "sendtoaddress".to_string(),
+            params: vec![address.to_string(), amount.as_btc().to_string()],
+        }
     }
 }
 
+#[derive(Debug, serde::Deserialize)]
+struct FundResponse {
+    result: sha256d::Hash,
```

Comment:
:wrench: 
```suggestion
    txid: sha256d::Hash,
```

Reply:
This is what bitcoind returns! I have no choice in the naming.